velocidi-sdk/src/main/kotlin/com/velocidi/HttpClient.kt

```diff
@@ -0,0 +1,63 @@
+package com.velocidi
+
+import android.util.Log
+import com.android.volley.RequestQueue
+import com.android.volley.Response
+import com.android.volley.toolbox.*
+import java.io.File
+import com.android.volley.toolbox.StringRequest
+import org.json.JSONObject
+import java.nio.charset.Charset
+
+
+class HttpClient {
+    private val cache = DiskBasedCache(File(Constants.CACHE_DIR), 1024 * 1024) // 1MB cap
+
+    private val network = BasicNetwork(HurlStack())
+
+    val requestQueue = RequestQueue(cache, network).apply {
+        start()
+    }
+
+    val headers = mutableMapOf(Pair("Content-Type", "application/json"))
+    val defaultParams = mutableMapOf<String,String>()
+
+
+    fun sendRequest(verb: Verb,url: String, payload: JSONObject? = null, listener: ResponseListener? = null) {
```

Comment:
What this function does is oddly specific for a "generic" HTTPClient. I would prefer to remove the logic of this function to a different class (maybe a `JsonRequestFactory`?) and have this function receive a `Request` and add it to the queue. Also, can we not avoid using the Nullables by replacing them with method overloading?

Reply:
As discussed offline, I thought that this only did the requests of Content-Type `application/json`. This is not the case. Never mind this!